src/wrapper-utils.js

```diff
@@ -0,0 +1,131 @@
+                exportFunction(descriptor[methodName], definedDescriptor, { defineAs: methodName })
+            }
+        })
+        UsedObjectInterface.defineProperty(usedObj, propertyName, definedDescriptor)
+    } else {
+        Object.defineProperty(object, propertyName, descriptor)
+    }
+}
+
+/**
+ * For each property defined on the object, update it with the target value.
+ */
+export function overrideProperty (name, prop) {
+    // Don't update if existing value is undefined or null
+    if (!(prop.origValue === undefined)) {
+        /**
+         * When re-defining properties, we bind the overwritten functions to null. This prevents
+         * sites from using toString to see if the function has been overwritten
+         * without this bind call, a site could run something like
+         * `Object.getOwnPropertyDescriptor(Screen.prototype, "availTop").get.toString()` and see
+         * the contents of the function. Appending .bind(null) to the function definition will
+         * have the same toString call return the default [native code]
+         */
+        try {
+            defineProperty(prop.object, name, {
+                // eslint-disable-next-line no-extra-bind
+                get: (() => prop.targetValue).bind(null)
+            })
+        } catch (e) {
+        }
+    }
+    return prop.origValue
+}
+
+/**
+ * add a fake toString() method to a wrapper function to resemble the original function
+ * @param {*} newFn
+ * @param {*} origFn
+ */
+function wrapToString (newFn, origFn) {
+    if (typeof newFn !== 'function' || typeof origFn !== 'function') {
+        return
+    }
+    newFn.toString = function () {
+        if (this === newFn) {
+            return functionToString.call(origFn)
+        } else {
+            return functionToString.call(this)
+        }
+    }
+}
+
+function getObjectFromPath (objPath, base = globalObj) {
+    const path = objPath.split('.')
+    if (path.length === 0) {
+        throw new Error('Invalid object path')
+    }
+
+    let object = base
+    for (const pathPart of path) {
+        if (!(pathPart in object)) {
+            // this happens if the object is not implemented in the browser
+            return
+        }
+        object = object[pathPart]
+    }
+    return object
+}
+
+/**
+ * Wrap a get/set or value property descriptor. Only for data properties. For methods, use wrapMethod(). For constructors, use wrapConstructor().
+ * @param {any} object - object or string object path within the global object
+ * @param {string} propertyName
+ * @param {Partial<PropertyDescriptor>} descriptor
+ * @returns {PropertyDescriptor|undefined} original property descriptor, or undefined if it's not found
+ */
+export function wrapProperty (object, propertyName, descriptor) {
+    if (!object) {
+        return
+    }
+    if (typeof object === 'string') {
+        object = getObjectFromPath(object)
+    }
```

Comment:
I think this is dead code even in your https://github.com/duckduckgo/content-scope-scripts/pull/527 patch, can we remove it along with the method too?

Reply:
👍 I removed it